code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm

```diff
@@ -189,6 +189,15 @@ GLOBAL_LIST_INIT(hugger_images_list,  list(
 	keybind_signal = COMSIG_XENOABILITY_DROP_ALL_HUGGER
 	use_state_flags = XACT_USE_LYING
 
+/datum/action/xeno_action/carrier_panic/give_action(mob/living/L)
+	. = ..()
+	RegisterSignal(owner, COMSIG_MOB_DEATH, .proc/action_activate)
```

Comment:
if it's a signal then it needs to have SIGNAL_HANDLER, and I really doubt it has it or should have it

Reply:
Can't because you are not able to `SIGNAL_HANDLER` an override which `action_activate` is.. so should I add it to the parent proc? Keep in mind this was your idea.